starknet_py/net/models/transaction.py

```diff
@@ -181,50 +175,6 @@ def calculate_hash(self, chain_id: int) -> int:
         )
 
 
-@dataclass(frozen=True)
-class DeclareV1(_DeprecatedAccountTransaction):
-    """
-    Represents a transaction in the Starknet network that is a declaration of a Starknet contract
-    class.
-    """
-
-    # The class to be declared, included for all methods involving execution (estimateFee, simulateTransactions)
-    contract_class: DeprecatedContractClass = field(
-        metadata={"marshmallow_field": fields.Nested(ContractClassSchema())}
-    )
-    # The address of the account contract sending the declaration transaction.
-    sender_address: int = field(metadata={"marshmallow_field": Felt()})
-
-    @property
-    def type(self) -> TransactionType:
-        return TransactionType.DECLARE
-
-    @marshmallow.post_dump
-    def post_dump(self, data: Dict[str, Any], **kwargs) -> Dict[str, Any]:
-        # Allowing **kwargs is needed here because marshmallow is passing additional parameters here
-        # along with data, which we don't handle.
-        # pylint: disable=unused-argument, no-self-use
-        return compress_program(data)
-
-    @marshmallow.pre_load
-    def pre_load(self, data: Dict[str, Any], **kwargs) -> Dict[str, Any]:
-        # pylint: disable=unused-argument, no-self-use
-        return decompress_program(data)
-
-    def calculate_hash(self, chain_id: int) -> int:
-        """
-        Calculates the transaction hash in the Starknet network.
-        """
-        return compute_declare_transaction_hash(
-            contract_class=self.contract_class,
-            chain_id=chain_id,
-            sender_address=self.sender_address,
-            max_fee=self.max_fee,
-            version=self.version,
-            nonce=self.nonce,
-        )
-
-
```

Comment:
Could you please explain why we remove this class? Other declareV1-related classes are kept, because we want to remove the ability for a user to send the declare v1 requests. But they still should be able to get the declare v1 transactions from the network.

Reply:
Correct me if I am wrong but `DeclareV1` was used for declaration only, but for collecting transactions from the block there is used [DeclareTransactionV1Schema](https://github.com/software-mansion/starknet.py/blob/kkawula/remove-sign-declare-v1/starknet_py/net/schemas/rpc/transactions.py#L176) that wasn't removed

Reply:
Right, thanks for clarification 👍